server/tokens/tokens.go

```diff
@@ -154,6 +159,17 @@ func (tb *TokenDetails) queryTokenDetails(conn *goclient.Client, byteCode string
 	lgr.Info("Analyzing contract byte code", zap.Stringer("address", tb.Contract),
 		zap.Strings("ercTypes", tb.ERCTypesSlice()), zap.Strings("functions", tb.FunctionsSlice()),
 		zap.Int("cachedContracts", count))
+	if _, ok := tb.ErcTypes[utils.Upgradeable]; ok {
+		if err := tb.queryTarget(conn, lgr); err != nil {
+			return err
+		}
+		//TODO ensure target loaded in db, and merge its types and funcs in to this set
```

Comment:
To finish #274, we need a hook here to get the types and funcs from the target (which may require a full load, if it's not in the DB), and then merge them in to this one. Then the other checks and the front end will include the full type and func sets. We can add this feature in a future PR.